test/lib/ansible_test/_internal/docker_util.py

```diff
@@ -195,10 +195,20 @@ def docker_images(args, image):
     except SubprocessError as ex:
         if 'no such image' in ex.stderr:
             stdout = ''  # podman does not handle this gracefully, exits 125
+        elif 'function "json" not defined' in ex.stderr:
+            # podman > 2 && < 2.2.0 breaks with --format {{json .}}, and requires --format json
+            # So we try this as a fallback. If it fails again, we just raise the exception and bail.
+            stdout, _dummy = docker_command(args, ['images', image, '--format', 'json'], capture=True, always=True)
         else:
             raise ex
-    results = [json.loads(line) for line in stdout.splitlines()]
-    return results
+
+    lines = stdout.splitlines()
+    if lines and lines[0][0] == '[':
```

Comment:
Usually, `str.startswith()` is considered more pythonic.

Reply:
I did it this way as a micro-optimization since we're only testing for one character, and indexing is an order of magnitude faster in the simple case:

```
rick@beryl /nvme/rick/dev/ansible/ansible $ python3 -m timeit '"foo".startswith("f")'
2000000 loops, best of 5: 133 nsec per loop

rick@beryl /nvme/rick/dev/ansible/ansible $ python3 -m timeit '"foo"[0] == "f"'
10000000 loops, best of 5: 20.3 nsec per loop
```

Though we do lose the advantage with double indexing like in this PR, so `stdout[0] = '['` would be better.
I was originally comparing the whole line and changed it at the last minute.

Reply:
JFYI it's probably better not to run tests on literals because it looks like CPython may optimize parsing `"foo"[0]` as `"f"` which affects the measurements. If you move the var declaration out of the test, the difference would be 2.5x vs 5x.
```python
Python 3.7.1 (default, Jan 28 2019, 08:25:13) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.1.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: %timeit "foo".startswith("f")                                         
129 ns ± 3.77 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
In [2]: %timeit "foo"[0] == "f"                                               
24.9 ns ± 1.47 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [3]: %timeit s="foo"; s[0] == "f"                                          
47 ns ± 2.4 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [4]: %timeit s="foo"; s.startswith("f")                                    
140 ns ± 9.35 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [5]: s = "foo"                                                               

In [6]: %timeit s.startswith("f")                                             
144 ns ± 5.74 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [7]: %timeit s[0] == "f"                                                   
56.7 ns ± 2.93 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```

Not that it matters on such a small code example. IIUC this code is not hit extensively during the test run, is it? It's definitely not a bottleneck. In such cases, I'd recommend prioritizing maintainability over low-level optimizations.